CRM/Price/BAO/LineItem.php

```diff
@@ -846,6 +846,15 @@ protected function getAdjustedFinancialItemsToRecord($entityID, $entityTable, $c
         }
         $financialItemsArray[$updateFinancialItemInfoValues['entity_id']] = $updateFinancialItemInfoValues;
       }
+      // INSERT a financial item to record 0 amount when to indicate price fee is not changed
```

Comment:
If this commit merely inserts a $0 financial item record on all line items that have not had their amounts changed during a transaction, I'm not sure it is worth adding unless it also resolve other things like the inappropriate proportional allocation. It also seems like a bit of a hack to solving that problem, but if it works I'll accept it.